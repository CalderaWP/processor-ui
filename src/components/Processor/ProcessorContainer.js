import React from 'react';
import PropTypes from 'prop-types';
import {RemoveProcessorButton} from "./RemoveProcessorButton";
import {OpenEditorButton} from "./OpenEditorButton";
import {Editor} from "./Editor";

export class ProcessorContainer extends React.PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			opacity: 1,
			hovered: false,
			isOpened: 'string' !== typeof props.processor.type
		};
		this.mouseEnter = this.mouseEnter.bind(this);
		this.mouseLeave = this.mouseLeave.bind(this);
		this.toggleOpen = this.toggleOpen.bind(this);
		this.onRemoveClick = this.onRemoveClick.bind(this);
		this.openButton = this.openButton.bind(this);
		this.onEdit = this.onEdit.bind(this);
	}

	/**
	 * Handler for the mouse enter event
	 */
	mouseEnter() {
		this.setState({hovered: true})
	}

	/**
	 * Handler for the mouse leave event
	 */
	mouseLeave() {
		this.setState({hovered: false})
	}

	/**
	 * Toggle if editor is open or not
	 */
	toggleOpen(){
		this.setState({isOpened:! this.state.isOpened });
	}

	/**
	 * Handle the remove processor action
	 */
	onRemoveClick(){
		this.props.onRemoveProcessor(this.props.processor.ID);
	}

	openButton(){
		let style = {
			opacity: 0.25
		};
		if ( this.state.hovered){
			style.opacity = 1
		}
		return(
			<OpenEditorButton
				style={style}
				isOpen={this.state.isOpened}
				onClick={this.toggleOpen}
			/>
		)
	}

	onEdit(newValues){
		this.props.onUpdateProcessor(newValues);
	}

	render() {
		let removeStyle = {
			opacity: 0.1
		};
		if ( this.state.hovered){
			removeStyle.opacity = 1
		}
		const labelAs = this.props.label ? this.props.label : this.props.ID;

		return (
			<div
				className={this.props.className}
				onMouseEnter={this.mouseEnter}
				onMouseLeave={this.mouseLeave}
			>

				<div>{labelAs}</div>
				{this.state.isOpened &&
					<Editor
						processor={this.props.processor}
						onUpdateProcessor={this.onEdit}
						form={this.props.form}
					/>
				}

				<div>
					{this.openButton()}
					<RemoveProcessorButton
						style={removeStyle}
						onClick={this.onRemoveClick}
					/>
				</div>
			</div>
		)
	}
}

ProcessorContainer.propTypes = {
	processor: PropTypes.object.isRequired,
	onRemoveProcessor: PropTypes.func.isRequired,
	onUpdateProcessor: PropTypes.func.isRequired,
	form: PropTypes.object.isRequired
};