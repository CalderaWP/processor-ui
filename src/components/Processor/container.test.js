import {shallow} from "enzyme/build/index";
import renderer from 'react-test-renderer';
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {ProcessorContainer} from "./ProcessorContainer";
Enzyme.configure({ adapter: new Adapter() });
describe( 'Processor Container component', () => {
	function handler(){

	}
	const form = {
		ID: 'CF1',
		name: 'Contact',
		field: {
			firstName: {

			},
			lastName: {

			}
		}
	};

	it( 'Passes snapshot test', () => {
		const component = renderer.create(
			<ProcessorContainer
				processor={{
					ID: 'p122',
				}}
				onUpdateProcessor={handler}
				onRemoveProcessor={handler}
				form={form}
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});
	it( 'Open if no type',  () =>{

		const wrapper = shallow(
			<ProcessorContainer
				processor={{
					ID: 'p122',
				}}
				onUpdateProcessor={handler}
				onRemoveProcessor={handler}
				form={form}
			/>
		);
		expect( wrapper.state('isOpened') ).toBe(true);
	});

	it( 'Not open if has type',  () =>{
		const wrapper = shallow(
			<ProcessorContainer
				processor={{
					ID: 'p122',
					type: 'redirect'
				}}
				onUpdateProcessor={handler}
				onRemoveProcessor={handler}
				form={form}
			/>
		);
		expect( wrapper.state('isOpened') ).toBe(false);
	});
});
