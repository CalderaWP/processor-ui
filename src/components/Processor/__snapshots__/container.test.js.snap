// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processor Container component Passes snapshot test 1`] = `
<div
  className={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div />
  <div
    className={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div />
    <div>
      Type: 
      <span
        className="processor-type"
      />
    </div>
    <select
      className="processor-type-chooser"
      onChange={[Function]}
      value={undefined}
    >
      <option>
        ---
      </option>
      <option
        value="email"
      >
        Email
      </option>
      <option
        value="redirect"
      >
        Redirect
      </option>
    </select>
  </div>
  <div>
    <button
      className={undefined}
      onClick={[Function]}
      style={
        Object {
          "opacity": 0.25,
        }
      }
    >
      <span>
        Close Editor
      </span>
    </button>
    <button
      className={undefined}
      onClick={[Function]}
      style={
        Object {
          "opacity": 0.1,
        }
      }
    >
      <span>
        Remove Processor
      </span>
    </button>
  </div>
</div>
`;
